test/server/TypeDBClusterRunner.java

```diff
@@ -18,34 +18,70 @@
 
 public class TypeDBClusterRunner extends TypeDBRunner {
 
-    private final Triple<Integer, Integer, Integer> ports;
-    private final List<Triple<Integer, Integer, Integer>> peerPorts;
+    public static final String OPT_ADDR = "--server.address";
+    private static final String OPT_INTERNAL_ADDR_ZMQ = "--server.internal-address.zeromq";
+    private static final String OPT_INTERNAL_ADDR_GRPC = "--server.internal-address.grpc";
+    public static final String OPT_PEERS_ADDR = "--server.peers.server-{peer-id}.address";
+    private static final String OPT_PEERS_INTERNAL_ADDR_ZMQ = "--server.peers.server-{peer-id}.internal-address.zeromq";
+    private static final String OPT_PEERS_INTERNAL_ADDR_GRPC = "--server.peers.server-{peer-id}.internal-address.grpc";
 
-    public TypeDBClusterRunner(Triple<Integer, Integer, Integer> ports, List<Triple<Integer, Integer, Integer>> peerPorts) throws InterruptedException, TimeoutException, IOException {
-        super();
-        this.ports = ports;
-        this.peerPorts = peerPorts;
+    private final Ports ports;
+    private final Set<Ports> peers;
+    private final Map<String, String> remainingServerOpts;
+
+    public static TypeDBClusterRunner create() throws InterruptedException, TimeoutException, IOException {
+        int port = ThreadLocalRandom.current().nextInt(40000, 60000);
+        Ports server = new Ports(port, port+1, port+2);
+        return TypeDBClusterRunner.create(server);
+    }
+
+    public static TypeDBClusterRunner create(Map<String, String> remainingServerOpts)
+            throws IOException, InterruptedException, TimeoutException {
+        int port = ThreadLocalRandom.current().nextInt(40000, 60000);
+        Ports server = new Ports(port, port+1, port+2);
+        return TypeDBClusterRunner.create(server, remainingServerOpts);
+    }
+
+    public static TypeDBClusterRunner create(Ports ports) throws IOException, InterruptedException, TimeoutException {
+        return TypeDBClusterRunner.create(ports, set());
     }
 
-    public TypeDBClusterRunner(Integer port) throws InterruptedException, TimeoutException, IOException {
-        this(triple(port, port + 1, port + 2), list(triple(port, port + 1, port + 2)));
+    public static TypeDBClusterRunner create(Ports ports, Set<Ports> peers)
+            throws IOException, InterruptedException, TimeoutException {
+        return TypeDBClusterRunner.create(ports, peers, map());
     }
 
-    public TypeDBClusterRunner() throws InterruptedException, TimeoutException, IOException {
-        this(ThreadLocalRandom.current().nextInt(40000, 60000));
+    public static TypeDBClusterRunner create(Ports ports, Map<String, String> remainingServerOpts)
+            throws IOException, InterruptedException, TimeoutException {
+        return new TypeDBClusterRunner(ports, set(), remainingServerOpts);
+    }
+
+    public static TypeDBClusterRunner create(Ports ports, Set<Ports> peers, Map<String, String> remainingServerOpts)
```

Comment:
Create a runner for a server within a cluster. The server will be started on the configured ports and options, communicating with the configured peers.